### PR TITLE
fix: Date parsing should support string format

### PR DESCRIPTION
**Pull request by wtrocki**

### CODE REVIEW

File: src/DateField.tsx [hunk 1]
```diff
@@ -5,7 +5,15 @@ import { connectField } from 'uniforms/es5';
...
 
 const DateConstructor = (typeof global === 'object' ? global : window).Date;
-const dateFormat = value => value && value.toISOString();
+const dateFormat = (value) => {
+  if(typeof value === "string"){
+    return value;
+  } else if(value.toISOString){
+    value && value.toISOString();
+  } else {
+    return undefined;
```
Comment (wtrocki) [inline src/DateField.tsx]: @kingsleyzissou Not sure how to handle this case. 
Data is invalid. Logging warning can be excessive here but if object is not string or date we cannot do much

File: src/DateField.tsx [hunk 2]
```diff
@@ -5,7 +5,17 @@ import { connectField } from 'uniforms/es5';
...
+  if(value){
+    if(typeof value === "string"){
+      return value;
+    } else if(value.toISOString){
+      value && value.toISOString();
+    } 
+  }
+  
+  return undefined
+}
```
Comment (machi1990) [inline src/DateField.tsx]:
```suggestion
const dateFormat = (value) => {
  if(value){
   return new Date(value).toISOString();   
  }
  
  return undefined
}
```

**Comment by wtrocki** [comment]
@machi1990 Not sure how this works but it works :D

**Review by kingsleyzissou — APPROVED** [review]

**Comment by kingsleyzissou** [comment]
@wtrocki think it's just checking if the value.toIsoString is defined. Took me a while to wrap my head around that too.

Good job @machi1990

**Comment by machi1990** [comment]
> @machi1990 Not sure how this works but it works :D

Just eleverages the Date constructors which accepts various date formats, the only time where it will stop working is the epoch time i.e timestamp `0`, because of the `if (value) ` statement, and undefined will be returned.

**Pull request merged**